### PR TITLE
Support partial request for media resource

### PR DESCRIPTION
**Pull request by MarschHuynh**
I am creating an API for record and playback audio. I use Eve as a main lib for my project.
Without partial request, my player in my phone (iPhone) can not work.

So I change a little code to make Eve support partial request for the media resource.

Hope that makes sense.

Thanks.

### CODE REVIEW

**Comment by Amedeo91** [comment]
@nicolaiarocci: what do you think? Personally I do not see test coverage...

**Comment by nicolaiarocci** [comment]
Hello and thanks for this. It is indeed a very cool feature. However, it would also need test coverage.

File: eve/endpoints.py [hunk 1]
```diff
@@ -174,28 +176,76 @@ def media_endpoint(_id):
...
+        size = file_.length
+        byte1, byte2 = 0, None
+
+        m = re.search('(\d+)-(\d*)', range_header)
+        g = m.groups()
+
+        if g[0]:
+            byte1 = int(g[0])
+        if g[1]:
+            byte2 = int(g[1])
```
Comment (peterdemin) [inline eve/endpoints.py]: `g[1]` might be empty string, which will break `int('')`:
```
>>> int('')
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
ValueError: invalid literal for int() with base 10: ''
```

Comment (MarschHuynh) [reply eve/endpoints.py]: @nicolaiarocci Yes, I will update the test as soon as possible.

@peterdemin Thank you for correcting me.

**Comment by nicolaiarocci** [comment]
This is still waiting on proper test coverage. Are you still planning to add it?

**Comment by nicolaiarocci** [comment]
Thanks for the commit update. I really want to merge this but I can't in its current state. Still needs documentantion and test coverage.

**Comment by nicolaiarocci** [comment]
Refactored, rebased, test-covered, and finally merged. See https://github.com/pyeve/eve/commit/9ee2b5baa97c361761ec73d7fa937ef82e9c310c

**Pull request closed without merging**